src/Valet/App.cs

```diff
@@ -74,6 +74,23 @@ public async Task<int> GetVersionAsync()
         return 0;
     }
 
+    public void CheckForUpdates()
+    {
+        try
+        {
+            var latestImageDigest = _dockerService.GetLatestImageDigestAsync(ValetImage, ValetContainerRegistry)?.Result;
+            var currentImageDigest = _dockerService.GetCurrentImageDigestAsync(ValetImage, ValetContainerRegistry).Result;
```

Comment:
```suggestion
    public async Task CheckForUpdatesAsync()
    {
        try
        {
            var latestImageDigest = await _dockerService.GetLatestImageDigestAsync(ValetImage, ValetContainerRegistry);
            var currentImageDigest = await _dockerService.GetCurrentImageDigestAsync(ValetImage, ValetContainerRegistry);
```

This should use the async/await primitives instead of directly calling `.Result` on a `Task`. It likely won't change much behavior here since this is essentially a single-threaded app, but if this were a multithreaded app, calling `.Result` could introduce deadlocks, hurt perf, etc. 😄

Reply:
We could also start parallel processes to get these 2 versions using async primitives too, which might help with the perf of this change.

```csharp
var latestImageDigestTask = _dockerService.GetLatestImageDigestAsync(ValetImage, ValetContainerRegistry);
var currentImageDigestTask = _dockerService.GetCurrentImageDigestAsync(ValetImage, ValetContainerRegistry);

await Task.WhenAll(latestImageDigestTask, currentImageDigestTask);

# Note: Calling .Result here is safe to do since we know the task has already completed and it won't be a blocking call
var latestImageDigest = latestImageDigestTask.Result;
var currentImageDigest = currentImageDigestTask.Result;
```

Reply:
Awesome. I did this, and unfortunately it still takes 4 seconds to run the update check. Which makes sense, since running each docker command manually results in: `docker image inspect` returns almost instantly (0.156s) and docker manifest inspect returns in 1.5s.

To be more clear as well, the '4 seconds' I mention also includes startup time for the application.

Reply:
Gotcha. It does seem to run in about half that time when the code is running as a release build. e.g.

```bash
$ dotnet publish src/Valet/Valet.csproj -c Release -r osx-x64 --self-contained -o dist/osx-x64 --no-restore -p:PublishSingleFile=true /p:IncludeNativeLibrariesForSelfExtract=true

$ ./dist/osx-x64/gh-valet version
```